src/services/telemetryService.ts

```diff
@@ -0,0 +1,113 @@
+import * as vscode from "vscode";
+import Analytics from "analytics-node";
+import { TelemetryEvent } from "../interfaces/telemetryEvent";
+import { Reporter } from "../reporter";
+import { Logger } from "../utils/logger";
+import { SegmentInitializer } from "../utils/segmentInitializer";
+import { TelemetryEventQueue } from "../utils/telemetryEventQueue";
+
+/* 
+  OPT_IN_STATUS == "true" if user Agreed
+  false when denied for telemetry data collection     
+*/
+const OPT_IN_STATUS = "redhat.telemetry.optInRequested";
+
+export namespace TelemetryService {
+  let analyticsObject: Analytics | undefined;
+
+  /* 
+    provides subscription to custom segment key.
+    fallback to default segment key if no key provided via API
+  */
+  export function subscribeTelemetryService(
+    context: vscode.ExtensionContext
+  ): boolean {
+    const CLIENT_SEGMENT_KEY: string | undefined = context.globalState.get(
+      "SEGMENT_KEY"
+    );
+    analyticsObject = SegmentInitializer.initialize(CLIENT_SEGMENT_KEY);
+    return analyticsObject != undefined;
+  }
+
```

Comment:
we'll prolly need to expose a startup(event) function, that will collect/add common data for all extensions. Can wait  for a later PR though

Reply:
okay sure, we can make an issue for this and take it up in the next PR.